### PR TITLE
Add flag `AnalysisState.executing_speculative_computations` to avoid flagging overflows during `invariant`

### PR DESCRIPTION
**Pull request by michael-schwarz**
As discussed in #1296

### CODE REVIEW

**Comment by michael-schwarz** [comment]
The question is whether we should undo #1406 in light of this more general solution.

File: src/analyses/baseInvariant.ml [hunk 1]
```diff
@@ -830,4 +830,12 @@ struct
             FD.top_of fk
         in
         inv_exp (Float ftv) exp st
+
+  let invariant ctx st exp tv =
+    (* The computations that happen here are not computations that happen in the programs *)
+    (* Any overflow during the forward evaluation will already have been flagged here *)
+    AnalysisState.executing_speculative_computations := true;
+    let res = invariant ctx st exp tv in
+    AnalysisState.executing_speculative_computations := false;
```
Comment (sim642) [inline src/analyses/baseInvariant.ml:839]: This should use `GobRef.wrap` at least. `invariant` can raise `Deadcode`.

**Comment by sim642** [comment]
The dune promotion stuff may have been due to trailing whitespace removal which should be disabled for `.t` files.

File: src/common/framework/analysisState.ml [hunk 2]
```diff
@@ -4,6 +4,12 @@
     This is set to true in control.ml before we verify the result (or already before solving if warn = 'early') *)
 let should_warn = ref false
 
+(** If this is true, any overflows happening in IntDomains will not lead to warnings being produced or
+    {!svcomp_may_overflow} being set to true. This is useful when, e.g., {!BaseInvariant.refine} executes computations that
```
Comment (sim642) [inline src/common/framework/analysisState.ml:8]:
```suggestion
    {!svcomp_may_overflow} being set to true. This is useful when, e.g., {!BaseInvariant.Make.invariant} executes computations that
```

**Review by sim642 — APPROVED** [review]

**Pull request merged**